ui/src/time_picker.tsx

```diff
@@ -0,0 +1,259 @@
+  visible?: B
+  /** True if the form should be submitted when the time is selected. */
+  trigger?: B
+  /** True if this is a required field. Defaults to False. */
+  required?: B
+  /** Specifies 12-hour or 24-hour time format. One of `12` or `24`. Defaults to `24`. */
+  hour_format?: S
+  /** The minimum allowed time value in hh:mm format. E.g.: '08:00', '13:30' */
+  min?: S
+  /** The maximum allowed time value in hh:mm format. E.g.: '15:30', '00:00' */
+  max?: S
+  /** Limits the available minutes to select from. One of `1`, `5`, `10`, `15`, `20`, `30` or `60`. Defaults to `1`. */
+  minutes_step?: U
+}
+
+const
+  css = stylesheet({
+    toolbar: {
+      paddingTop: 16,
+      paddingLeft: 16
+    },
+    toolbarTime: {
+      fontSize: 26,
+      cursor: 'pointer',
+      $nest: {
+        '&:hover': {
+          backgroundColor: cssVar('$neutralLighter'),
+          color: cssVar('$neutralDark')
+        }
+      }
+    },
+    toolbarText: { fontSize: 26 },
+    toolbarLabel: { maxWidth: '70%' }
+  }),
+  popoverProps: Partial<PopperProps> | undefined = {
+    placement: 'bottom-start',
+    sx: {
+      '& .MuiPaper-root': {
+        borderRadius: '2px',
+        boxShadow: `${cssVar('$text1')} 0px 6.4px 14.4px 0px, ${cssVar('$text2')} 0px 1.2px 3.6px 0px`
+      },
+      '& .MuiTypography-root, [role=listbox]>span': {
+        fontFamily: '"Inter", -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue", sans-serif',
+      }
+    }
+  }
+
+type ToolbarProps = { time: S, setOpenView: (view: CalendarOrClockPickerView) => void, label: S | undefined, switchAmPm: () => void }
+
+const
+  ThemeProvider = React.lazy(() => import('@mui/material/styles').then(({ ThemeProvider }) => ({ default: ThemeProvider }))),
+  LocalizationProvider = React.lazy(() => import('@mui/x-date-pickers').then(({ LocalizationProvider }) => ({ default: LocalizationProvider }))),
+  TimePicker = React.lazy(() => import('@mui/x-date-pickers').then(({ TimePicker }) => ({ default: TimePicker }))),
+  parseTimeStringToDate = (time: S) => new Date(`2000-01-01T${time.slice(0, 5)}:00`),
+  useMuiTheme = (themeObj: ThemeOptions) => {
+    const [theme, setTheme] = React.useState<Theme>()
+    React.useEffect(() => {
+      import('@mui/material/styles')
+        .then(({ createTheme }) => setTheme(createTheme(themeObj)))
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [])
+    return [theme]
+  },
+  useAdapterDateFns = () => {
+    const [AdapterDateFns, setAdapterDateFns] = React.useState<typeof DateFnsUtils | null>()
+    React.useEffect(() => {
+      import('@mui/x-date-pickers/AdapterDateFns')
+        .then(({ AdapterDateFns }) => { setAdapterDateFns(() => AdapterDateFns) })
+    }, [])
+    return [AdapterDateFns]
+  },
+  useDateFns = () => {
+    const [dateFns, setDateFns] = React.useState<any>()
+    React.useEffect(() => {
+      import('date-fns')
```

Comment:
Can we import a single function instead of whole style module here?

```suggestion
      import('date-fns/esm/format')
```

Reply:
We can, but why not `import('date-fns/format')`?

Reply:
If the bundle stays the same size, then `date-fns/format` is fine as well.

Reply:
Although `esm` shall be smaller, I assume you will hit the same problem as [here](https://github.com/h2oai/wave/pull/1609#discussion_r970393299) so just add TODO.

Reply:
Seems there is a `6kB` difference. I'll add TODO as well.